sdk/src/wallet/wallet/operations/stronghold_backup/mod.rs

```diff
@@ -175,30 +176,9 @@ impl Wallet {
             }
         }
 
-        // store new data
+        // save potential restored accounts to db
         #[cfg(feature = "storage")]
         {
-            let wallet_builder = WalletBuilder::new()
-                .with_secret_manager_arc(self.secret_manager.clone())
-                .with_storage_path(
-                    &self
-                        .storage_options
-                        .storage_path
-                        .clone()
-                        .into_os_string()
-                        .into_string()
-                        .expect("can't convert os string"),
-                )
-                .with_client_options(self.client_options().await)
-                .with_coin_type(self.coin_type.load(Ordering::Relaxed));
-            // drop secret manager, otherwise we get a deadlock in save_wallet_data
-            drop(secret_manager);
-            self.storage_manager
-                .read()
-                .await
-                .save_wallet_data(&wallet_builder)
-                .await?;
-            // also save account to db
```

Comment:
Why is this removed?

Reply:
Because it's saved when we set the client options

Reply:
Ok but this was always done, while `set_client_options` is conditionally done

Reply:
Oh right, reverted